### DISCUSSION TITLE
Recommended pattern for combining multiple apollo mutation|query elements?

### DISCUSSION THREAD

**Discussion by iambumblehead** [Akryum/vue-apollo#666] [Q&A]
For example, if a component needs to use mutations 'createData', 'updateData' and 'deleteData' one might include three nested ApolloMutation elements. Is there a nice way to combine the ApolloMutation elements to one element?

**Comment by renestalder** [comment]
That's also the part I don't get. The documentation always only shows the request on one resolver, but never with multiple. Can't get something like the following working as the name matching will omit the property `category`:

```js
    paginatedProducts: {
      query: gql`
        query paginatedProducts($categoryIds: [ID!], $categoryId: ID!) {
          paginatedProducts(categoryArray: $categoryIds) {
            products {
              articleNumber
            }
            pagination {
              total
            }
          }
          category(id: $categoryId) {
            nameDe
          }
        }
      `,
```

I can separate queries in the smart query definition, but this adds unnecessary back-end server calls.

**Comment by Akryum** [comment]
@iambumblehead You get the mutation result data with everything returned by the mutation.

@renestalder use the `result` hook.

**Comment by renestalder** [comment]
@Akryum Thanks for the quick response. Kind of assumed that solution.
